boot/freeldr/freeldr/cmdline.c

```diff
@@ -36,6 +36,7 @@ CmdLineParse(IN PCSTR CmdLine)
     CmdLineInfo.DefaultOs = NULL;
     CmdLineInfo.TimeOut = -1;
 
+    if (!CmdLine) return;
```

Comment:
Why do you get a NULL pointer passed there? (And why weren't there any problem for non-UEFI builds?)

Reply:
https://github.com/alula/reactos/blob/arm64bringup/boot/freeldr/freeldr/arch/uefi/uefildr.c#L28

I literally pass null into this

Reply:
`BootMain()` gets called with a `NULL` in uefildr.c: https://github.com/reactos/reactos/blob/f2a58733e85158a8325e657e9cdfc32f35d516ba/boot/freeldr/freeldr/arch/uefi/uefildr.c#L28

`NULL` in strstr() below is UB and caused crashes when I ran it.

Reply:
Then don't pass NULL, but respect the function's "contract" (it says "IN PCSTR....", not "IN PCSTR .... OPTIONAL" or equivalent in the SAL2 annotation)
See for example:
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/arch/arm/entry.c;hb=945ee4b2a47402e99beb8a8e877c32e6cf81bbcd#l19
All other callers also pass a valid non-NULL pointer:
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/arch/i386/entry.S;hb=945ee4b2a47402e99beb8a8e877c32e6cf81bbcd#l90
(with BootMain calling CmdLineParse https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/freeldr.c;hb=945ee4b2a47402e99beb8a8e877c32e6cf81bbcd#l43 )

Reply:
I’ll change this in master later I guess then.